vendor/github.com/Azure/azure-sdk-for-go/arm/appinsights/components.go

```diff
@@ -98,7 +98,9 @@ func (client ComponentsClient) CreateOrUpdatePreparer(resourceGroupName string,
 // CreateOrUpdateSender sends the CreateOrUpdate request. The method will close the
 // http.Response Body if it receives an error.
 func (client ComponentsClient) CreateOrUpdateSender(req *http.Request) (*http.Response, error) {
-	return autorest.SendWithSender(client, req)
+	return autorest.SendWithSender(client,
+		req,
+		azure.DoRetryWithRegistration(client.Client))
```

Comment:
we've got a variable [to opt-out of Resource Provider registration](https://www.terraform.io/docs/providers/azurerm/index.html#skip_provider_registration) - which we use to skip checking for the Resource Provider Registration.

This method will attempt to register the Resource Provider in the case that it's unregistered, which means it has a different behaviour to the variable listed above - however I'm going to open an issue on Go-AutoRest asking for a means of failing when Resource Providers aren't registered, rather than registering - so that we can still support this use-case.

I've opened an issue on Go-AutoRest about this: https://github.com/Azure/go-autorest/issues/210

Reply:
There's a feature toggle for this coming in v12 - but there's no harm in shipping this as-is.

Running the test suite now..

Reply:
Tests seem good 👍